### DISCUSSION TITLE
RabbitMQ as queue provider for tasks

### DISCUSSION THREAD

**Discussion by fejo** [Netflix/conductor-community#100] [Ideas]
Hi,

we are considering using Conductor as a worklfow orchestrator but do not want to use the deprecated dyno-queues or any sql database for task queues. We are already using RabbitMQ as a message broker and it makes sense for us to use it as a Conductor task queue provider too. Do you have any plans to officially support RabbitMQ as a task queue provider? Have you ever considered this option?

Thanks

**Comment by pjanzen** [comment]
I cannot answer your question but rabbitmq is configurable in case you cannot find the settings for it.

You can find the file [here](https://github.com/Netflix/conductor/tree/main/server/src/main/resources)

```
#Following properties set for using AMQP events and tasks with conductor:
#(To enable support of AMQP queues)
#conductor.event-queues.amqp.enabled=true

# Here are the settings with default values:
#conductor.event-queues.amqp.hosts=<rabbitmq serverip>
#conductor.event-queues.amqp.username=<username>
#conductor.event-queues.amqp.password=<password>

#conductor.event-queues.amqp.virtualHost=/
#conductor.event-queues.amqp.port=5672
#conductor.event-queues.amqp.useNio=false
#conductor.event-queues.amqp.batchSize=1
#conductor.event-queues.amqp.pollTimeDuration=100ms
#conductor.event-queues.amqp.queueType=classic
#conductor.event-queues.amqp.sequentialMsgProcessing=true
#conductor.event-queues.amqp.connectionTimeoutInMilliSecs=180000
#conductor.event-queues.amqp.networkRecoveryIntervalInMilliSecs=5000
#conductor.event-queues.amqp.requestHeartbeatTimeoutInSecs=30
#conductor.event-queues.amqp.handshakeTimeoutInMilliSecs=180000
#conductor.event-queues.amqp.maxChannelCount=5000
#conductor.event-queues.amqp.limit=50
#conductor.event-queues.amqp.duration=1000
#conductor.event-queues.amqp.retryType=REGULARINTERVALS

#conductor.event-queues.amqp.useExchange=true( exchange or queue)
#conductor.event-queues.amqp.listenerQueuePrefix=myqueue
# Use durable queue ?
#conductor.event-queues.amqp.durable=false
# Use exclusive queue ?
#conductor.event-queues.amqp.exclusive=false
# Enable support of priorities on queue. Set the max priority on message.
# Setting is ignored if the value is lower or equals to 0
#conductor.event-queues.amqp.maxPriority=-1
```

**Reply by fejo** [reply]
Hi pjanzen, thank you for your answer. 

This settings is used for event queues configuration.
 
We want to use RabbitMQ as a persistence layer for task queues, i.e. implement this interface https://github.com/Netflix/conductor/blob/main/core/src/main/java/com/netflix/conductor/dao/QueueDAO.java and use RabbitMQ as a backend (instead of dyno-queues or any sql database).

**Comment by apanicker-nflx** [comment]
@fejo Conductor task queue API is certainly extensible to integrate with RabbitMQ. However, this is not on our roadmap. I will move this thread into conductor-community repo and see if anyone in the community is interested in this.
Additionally, we would welcome a contribution from you if you were to implement this functionality.
